cql3/statements/truncate_statement.cc

```diff
@@ -39,12 +39,7 @@ std::unique_ptr<prepared_statement> truncate_statement::prepare(data_dictionary:
     return std::make_unique<prepared_statement>(::make_shared<truncate_statement>(*this));
 }
 
-bool truncate_statement::depends_on_keyspace(const sstring& ks_name) const
-{
-    return false;
-}
-
-bool truncate_statement::depends_on_column_family(const sstring& cf_name) const
+bool truncate_statement::depends_on(std::string_view ks_name, std::optional<std::string_view> cf_name) const
 {
```

Comment:
Note: I think truncates do depend on the table, for example it might get dropped.

Reply:
I also think so, I was under the impression that some of those were skipped because they couldn't be prepared but
TRUNCATE can be prepared. I will try to go over all queries one by one to see if I missed more of them that should
depend on table or keyspace.